web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EnrollEksCluster.tsx

```diff
@@ -0,0 +1,421 @@
+import styled from 'styled-components';
+import { FetchStatus } from 'design/DataTable/types';
+import { Danger } from 'design/Alert';
+
+import useAttempt from 'shared/hooks/useAttemptNext';
+import { ToolTipInfo } from 'shared/components/ToolTip';
+import { getErrMessage } from 'shared/utils/errorType';
+
+import { EksMeta, useDiscover } from 'teleport/Discover/useDiscover';
+import {
+  Regions,
+  integrationService,
+  AwsEksCluster,
+} from 'teleport/services/integrations';
+
+import { AwsRegionSelector } from 'teleport/Discover/Shared/AwsRegionSelector';
+import { ConfigureIamPerms } from 'teleport/Discover/Shared/Aws/ConfigureIamPerms';
+import { isIamPermError } from 'teleport/Discover/Shared/Aws/error';
+import { AgentStepProps } from 'teleport/Discover/types';
+import useTeleport from 'teleport/useTeleport';
+
+import { useJoinTokenSuspender } from 'teleport/Discover/Shared/useJoinTokenSuspender';
+import { generateCmd } from 'teleport/Discover/Kubernetes/HelmChart/HelmChart';
+import { Kube } from 'teleport/services/kube';
+
+import { ActionButtons, Header, ResourceKind } from '../../Shared';
+
+import { ClustersList } from './EksClustersList';
+import { ManualHelmDialog } from './ManualHelmDialog';
+import { EnrollmentDialog } from './EnrollmentDialog';
+import { AgentWaitingDialog } from './AgentWaitingDialog';
+
+type TableData = {
+  items: CheckedEksCluster[];
+  fetchStatus: FetchStatus;
+  startKey?: string;
+  currRegion?: Regions;
+};
+
+const emptyTableData: TableData = {
+  items: [],
+  fetchStatus: 'disabled',
+  startKey: '',
+};
+
+// CheckedEksCluster is a type to describe that a
+// Kube cluster has been checked with the backend
+// whether or not a kube server already exists for it.
+export type CheckedEksCluster = AwsEksCluster & {
+  kubeServerExists?: boolean;
+};
+
+type EKSClusterEnrollmentState = {
+  status: 'notStarted' | 'enrolling' | 'awaitingAgent' | 'success' | 'error';
+  error?: string;
+};
+
+export function EnrollEksCluster(props: AgentStepProps) {
+  const { agentMeta, emitErrorEvent } = useDiscover();
+  const { attempt: fetchClustersAttempt, setAttempt: setFetchClustersAttempt } =
+    useAttempt('');
+
+  const [tableData, setTableData] = useState<TableData>({
+    items: [],
+    startKey: '',
+    fetchStatus: 'disabled',
+  });
+  const [selectedCluster, setSelectedCluster] = useState<CheckedEksCluster>();
+  const [selectedRegion, setSelectedRegion] = useState<Regions>();
+  const [confirmedCluster, setConfirmedCluster] = useState<Kube>();
+  const [enrollmentState, setEnrollmentState] =
+    useState<EKSClusterEnrollmentState>({
+      status: 'notStarted',
+    });
+  const [isAppDiscoveryEnabled, setAppDiscoveryEnabled] =
+    useState<boolean>(true);
+  const [isAgentWaitingDialogShown, setIsAgentWaitingDialogShown] =
+    useState<boolean>(false);
+  const [isManualHelmDialogShown, setIsManualHelmDialogShown] =
+    useState<boolean>(false);
```

Comment:
when you set a initial value like this, typescript can infer the type for you, so we dont have to explicitly type it
```suggestion
  const [isAppDiscoveryEnabled, setAppDiscoveryEnabled] =
    useState(true);
  const [showAgentWaitingDialog, setShowAgentWaitingDialog] =
    useState(false);
  const [showManualHelmDialog, setShowManualHelmDialog] =
    useState(false);
```

Reply:
https://github.com/gravitational/teleport/pull/36634/commits/e7219fcb9f22049dbca13e6ab28a01ca8269e348